libsolidity/analysis/FunctionCallGraph.cpp

```diff
@@ -140,7 +141,15 @@ bool FunctionCallGraphBuilder::visit(EmitStatement const& _emitStatement)
 
 bool FunctionCallGraphBuilder::visit(Identifier const& _identifier)
 {
-	if (auto const* callable = dynamic_cast<CallableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+	if (auto const* variable = dynamic_cast<VariableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+	{
+		if (variable->isConstant())
+		{
+			solAssert(variable->isStateVariable(), "");
```

Comment:
Does the following fail the assertion?

```
bytes4 constant x = C.f.selector
contract C { function f() public {} }
```

Reply:
I think it won't but I'll add this case to the test to make sure.

Reply:
Test added. As expected, this case works fine - simply because we're not visiting the file-level constants when building the graph.

Reply:
Actually something is triggering this assertion (though I don't think it's your example). I see on CircleCI that there are some tests that failed because of it. I'll investigate.

Reply:
Looks like you were right, though your example won't trigger it on its own. It also needs something that references the constant, otherwise it's not visited.

I added an extra check for `variable->isFileLevelVariable()` to the assertion and with this it passes tests for me.

But anyway, good catch!